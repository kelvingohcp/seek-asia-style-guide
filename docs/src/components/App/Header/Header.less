@import (reference) "~seek-asia-style-guide/theme";

.headerBlock {
  background: @sk-blue;
}

.fixedHeaderBlock {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 10;
}

.logoLink {
  display: inline-block;
}

.logo {
  width: 118px;
  height: 31px;
  position: relative;
  top: 3px;
  margin-bottom: 3px;
}

.title {
  .yellingTextResponsive();

  text-transform: lowercase;
  position: absolute;
  color: @sk-white;
  text-decoration: none;
  top: -2px;
  left: 126px;
  width: 150px;
}

.sectionContent {
  display: flex;
  align-items: center;
  justify-content: space-between;
  min-height: @grid-base * 10;
}

.hamburgerText {
  overflow: hidden;
  width: 0;
  height: 0;
  text-indent: 999px;
  position: absolute;
}

.checkbox {
  position: absolute;
  width: 0;
  height: 0;
  opacity: 0;
}

.bars {
  display: block;
  cursor: pointer;
  width: @grid-gutter-width * 2;
  height: @grid-row-height * 7;
  position: relative;
  transform-origin: 100% 50%;
  transform: scale(0.7);
}

@barHeight: 6px;
.bar() {
  height: @barHeight;
  width: 100%;
  background: @sk-white;
  position: absolute;
  transition: all 0.1s ease;
}

.bar1 {
  .bar();
  top: 0;
  .checkbox:checked ~ .bars & {
    transform: translateY(11px) rotate(45deg);
  }
}

.bar2 {
  .bar();
  top: 50%;
  margin-top: @barHeight * -0.5;
  .checkbox:checked ~ .bars & {
    opacity: 0;
  }
}

.bar3 {
  .bar();
  bottom: 0;
  .checkbox:checked ~ .bars & {
    transform: translateY(-11px) rotate(-45deg);
  }
}

.menu {
  background: @sk-blue;
  color: @sk-white;
  position: fixed;
  z-index: 10;
  top: @grid-row-height * 16;
  left: 0;
  right: 0;
  bottom: 0;
  padding-bottom: @grid-row-height * 20;
  overflow: auto;
  display: none;

  .checkbox:checked ~ & {
    display: block;
  }
}

.componentList {
  display: flex;
  flex-wrap: wrap;
}

.componentListItem {
  flex: 0 0 (@grid-base * 64);
}

.componentSectionTitle {
  padding-top: @grid-base * 8;
  border-bottom: medium solid rgba(255, 255, 255, 0.2);
}

.link {
  color: white;
  text-decoration: none;
  display: inline-block;
  margin-bottom: @grid-row-height * 2;
  &:hover {
    box-shadow: inset 0 -2px 0 @sk-white;
  }
}
