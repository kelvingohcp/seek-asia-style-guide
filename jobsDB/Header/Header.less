@import (reference) "~seek-asia-style-guide/theme";

.primaryNavLink,
.primaryNavLink:visited {
  color: @dbBrand;
}

.primaryNavLink:hover {
  color: darken(@dbBrand, 10%);
}

.activeActionTrayIcon {
  color: @dbBrand;
}

.menuIcon {
  color: @dbBrand;
  fill: currentColor;
}
