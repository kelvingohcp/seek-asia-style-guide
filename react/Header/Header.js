import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import ScreenReaderOnly from '../ScreenReaderOnly/ScreenReaderOnly';
import { Text, PageBlock } from 'seek-asia-style-guide/react';
import Menu from './components/Menu/Menu';
import ActionTray from './components/ActionTray/ActionTray';
import styles from './Header.less';
import UserAccount from './components/UserAccount/UserAccount';
import { AUTHENTICATED, UNAUTHENTICATED, AUTH_PENDING } from 'seek-asia-style-guide/react/private/authStatusTypes';
import _get from 'lodash/get';
import { Icon } from 'seek-asia-style-guide/react';

const currentLocale = ({ title, ItemIcon }) => {
  return (
    <span className={styles.currentLocale}>
      <ItemIcon className={styles.localeIcon} />
      <Text whispering>{title}</Text>
    </span>
  );
};

const defaultLinkRenderer = props => (<a {...props} />);

currentLocale.propTypes = {
  title: PropTypes.string,
  ItemIcon: PropTypes.func
};

const renderPrimaryNavLinks = ({ brandStyles }, links, _style, linkRenderer) => {
  const primaryNavLinks = (links && links.map) ?
    links.map((link, index) => {
      return (
        <span key={index} className={link.hideInDesktop ? styles.hideInDesktop : styles.primaryNavLinkWrapper}>
          {
            linkRenderer({
              href: link.url,
              className: classnames(styles.primaryNavLink, brandStyles.primaryNavLink),
              children: <Text data-automation={link.automationId}>{link.title}</Text>
            })}
        </span>
      );
    }) : [];
  return (
    <div className={_style}>
      {primaryNavLinks}
    </div>
  );
};

renderPrimaryNavLinks.propTypes = {
  links: PropTypes.array,
  brandStyles: PropTypes.object.isRequired,
  linkRenderer: PropTypes.func
};

export default class Header extends Component {
  constructor() {
    super();

    this.state = {
      menuOpen: false
    };
    this.handleResize = this.handleResize.bind(this);
  }
  componentDidMount() {
    window.addEventListener('resize', this.handleResize);
  }
  componentWillUnmount() {
    window.addEventListener('resize', null);
  }
  handleResize() {
    if (this.state.menuOpen && window.innerWidth > 760) {
      this.setState({ menuOpen: false });
    }
  }

  handleToggleMenu() {
    const { onMenuOpen } = this.props;
    const { menuOpen } = this.state;
    if (!menuOpen) {
      onMenuOpen();
    }
    this.setState({ menuOpen: !this.state.menuOpen });
  }

  render() {
    const {
      LogoComponent,
      logoProps,
      homeUrl,
      activeTab,
      links,
      locales,
      headerMessage,
      menuMessage,
      brandStyles,
      country,
      actionTrayProps,
      employerSite,
      loginAvailable = false,
      authenticationStatus,
      userName,
      userAccMenuItems,
      baseUrl,
      linkRenderer,
      hideNavMenu
    } = this.props;
    const { showTray } = actionTrayProps;
    const menuOpen = this.state.menuOpen;
    const shouldShowMenuWithSpace = !showTray;

    return (
      <header className={styles.root}>
        <PageBlock>
          <div className={classnames({ [styles.primaryNavNoLogin]: !loginAvailable }, styles.primaryNav)}>
            <h1 className={styles.logo}>
              {
                linkRenderer({
                  className: styles.logoLink,
                  href: homeUrl,
                  children: <ScreenReaderOnly>{_get(headerMessage, 'homeUrlText')}</ScreenReaderOnly>
                })
              }
              <LogoComponent {...logoProps} />
            </h1>

            {!hideNavMenu && renderPrimaryNavLinks({ brandStyles }, links, styles.primaryNavLinksWrapper, linkRenderer)}
            {loginAvailable && <div className={styles.secondaryNav} />}
            <UserAccount
              userName={userName}
              loginAvailable={loginAvailable}
              authenticationStatus={authenticationStatus}
              brandStyles={brandStyles}
              baseUrl={baseUrl}
              userAccMenuItems={userAccMenuItems}
              headerMessage={headerMessage}
              linkRenderer={linkRenderer}
            />
            {!showTray &&
            <div onClick={this.handleToggleMenu.bind(this)} className={styles.menuToggle}>
              <Icon type="menu" className={classnames(styles.svg, { [brandStyles.activeActionTrayIcon]: menuOpen })} size="normal" />
            </div>}
          </div>
          <ActionTray
            country={country}
            {...actionTrayProps}
            brandStyles={brandStyles}
            headerMessage={headerMessage}
            menuOpen={menuOpen}
            handleToggleMenu={this.handleToggleMenu.bind(this)}
            activeTab={activeTab}
            baseUrl={baseUrl}
            linkRenderer={linkRenderer}
          />
          <Menu
            shouldShowMenu={menuOpen}
            shouldShowMenuWithSpace={shouldShowMenuWithSpace}
            headerMessage={headerMessage}
            menuMessage={menuMessage}
            links={links}
            locales={locales}
            brandStyles={brandStyles}
            loginAvailable={loginAvailable}
            employerSite={employerSite}
            authenticationStatus={authenticationStatus}
            userName={userName}
            linkRenderer={linkRenderer}
          />
        </PageBlock>
      </header>
    );
  }
}

Header.propTypes = {
  loginAvailable: PropTypes.bool,
  LogoComponent: PropTypes.func.isRequired,
  logoProps: PropTypes.object,
  activeTab: PropTypes.string,
  homeUrl: PropTypes.string,
  links: PropTypes.array,
  locales: PropTypes.array.isRequired,
  country: PropTypes.string.isRequired,
  headerMessage: PropTypes.object.isRequired,
  menuMessage: PropTypes.object.isRequired,
  brandStyles: PropTypes.object.isRequired,
  actionTrayProps: PropTypes.object,
  employerSite: PropTypes.bool,
  linkRenderer: PropTypes.func,
  onMenuOpen: PropTypes.func,
  authenticationStatus: PropTypes.oneOf([
    AUTHENTICATED,
    UNAUTHENTICATED,
    AUTH_PENDING
  ]),
  userName: PropTypes.string,
  userAccMenuItems: PropTypes.array,
  baseUrl: PropTypes.string,
  hideNavMenu: PropTypes.bool
};

Header.defaultProps = {
  linkRenderer: defaultLinkRenderer,
  authenticationStatus: UNAUTHENTICATED,
  homeUrl: '/',
  onMenuOpen: () => {},
  hideNavMenu: false
};
