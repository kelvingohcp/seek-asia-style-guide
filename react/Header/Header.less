@import (reference) "~seek-asia-style-guide/theme";

.root {
  width: 100%;
  background-color: @saWhite;
  z-index: @z-index-header;
}

._cfs_container_width() {
  margin-left: auto;
  margin-right: auto;
}

.externalNav {
  display: none;
  ._cfs_container_width;
  @media @desktop {
    display: flex;
    flex-direction: row-reverse;
    justify-content: space-between;
    align-items: center;
    height: @grid-base * 4;
    padding: 0 @grid-base * 6;
  }
  &::before,
  &::after {
    display: none !important;
  }
}

.currentLocale {
  display: flex;
  margin-top: @grid-base*4;
  align-items: center;
}

.localeIcon {
  margin: @grid-base @grid-base 0 0;
}

.primaryNav {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: center;
  padding: 1px @grid-base * 2 0 @grid-base * 2; /* 1px padding fixes baseline for align-centered items in this container */
  height: @grid-base * 12;
  justify-content: center;
  ._cfs_container_width;
  @media @desktop {
    justify-content: space-between;
    padding: 1px @grid-base * 6 0 @grid-base * 4;
  }
}

.primaryNavNoLogin {
  &:extend(.primaryNav);
  justify-content: center;
  @media @desktop {
    justify-content: flex-start;
  }
}

.primaryNavNoCountry {
  @media @desktop {
    height: @grid-base * 14;
  }
}

.primaryNavWithCountry {
  @media @desktop {
    height: @grid-base * 16;
  }
}

.primaryNavLinksWrapper {
  display: none;
  @media @desktop {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    align-items: center;
    padding-left: @grid-base * 8;
  }
}

.primaryNavLinkWrapper {
  flex-grow: 1;
  cursor: pointer;
  margin-right: @grid-base * 8;
  &:last-child {
    margin-right: 0;
  }
}

.primaryNavLink {
  text-decoration: none;
  display: inline-block;
}

.primaryNavLink .icon {
  margin: auto @grid-base * 2;
}

.secondaryNav {
  margin-left: auto;
  display: none;
  min-width: max-content;
  justify-content: space-between;
  @media @desktop {
    display: flex;
  }
}

.hideInDesktop {
  @media @desktop {
    display: none;
  }
}

.logo {
  position: relative;
  top: 3px;
  @media @desktop {
    display: flex;
  }
}

.logoSvg {
  height: 39px;
  width: 105px;
  @media @desktop {
    height: 51px;
    width: 129px;
  }
}

.logoLink {
  position: absolute;
  top: 0;
  text-indent: 100%;
  overflow: hidden;
  width: 100%;
  height: 100%;
}
