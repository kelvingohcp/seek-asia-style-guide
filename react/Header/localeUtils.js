export const sortCurrentLocaleToTop = ({ locales, country, language }) => {
  const currentLocale = locales.filter(locale => {
    return locale.country === country && locale.language === language;
  });

  const otherLocales = locales.filter(locale => {
    return !(locale.country === country && locale.language === language);
  });

  return [
    ...currentLocale,
    ...otherLocales
  ];
};
