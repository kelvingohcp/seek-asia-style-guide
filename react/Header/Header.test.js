import React from 'react';
import { shallow } from 'enzyme';

import Header from './Header';

describe('Header', () => {
  const FakeComponent = () => {
    return (<div>I'm an image of some kind</div>);
  };

  const defaultProps = {
    brandStyles: {
      primaryNavLink: 'primaryNavLink',
      employerLink: 'employerLink'
    },
    messages: {
      'header.employerSiteUrl': 'http://brand.seekasia.com',
      'header.employerSiteTitle': 'click here',
      'header.employerLinkPrefix': 'Awesome employers should '
    },
    locales: [{
      title: 'Hong Kong (English)',
      ItemIcon: FakeComponent,
      url: 'https://hk.jobsdb.com/hk',
      language: 'en',
      country: 'hk'
    }, {
      title: 'Indonesia (English)',
      ItemIcon: FakeComponent,
      url: 'https://id.jobsdb.com/id',
      language: 'en',
      country: 'id'
    }],
    LogoComponent: FakeComponent,
    country: 'hk',
    language: 'en'
  };

  const mockLinks = [
    { title: 'Home', url: 'http://seekasia.com/home', ItemIcon: FakeComponent },
    { title: 'MyPortal', url: 'http://seekasia.com/myPortal', ItemIcon: FakeComponent },
    { title: 'Resources', url: 'http://seekasia.com/resources', ItemIcon: FakeComponent },
    { title: 'CareerInsights', url: 'http://seekasia.com/career-insights', ItemIcon: FakeComponent }
  ];

  it('should render with default props', () => {
    const wrapper = shallow(<Header {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with login available', () => {
    const testProps = {
      ...defaultProps,
      loginAvailable: true
    };
    const wrapper = shallow(<Header {...testProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with links list', () => {
    const testProps = {
      ...defaultProps,
      links: mockLinks
    };
    const wrapper = shallow(<Header {...testProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should pass through optional props', () => {
    const testProps = {
      ...defaultProps,
      links: mockLinks,
      activeTab: 'ACTIVE_TAB_HOME',
      logoProps: { country: 'my' },
      more: [{ title: 'Space Piracy Jobs', url: 'http://seekasia.com/space-piracy' }],
      actionTrayProps: {
        showMenu: false
      }
    };
    const wrapper = shallow(<Header {...testProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
