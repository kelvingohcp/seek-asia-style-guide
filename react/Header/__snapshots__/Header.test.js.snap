// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should pass through optional props 1`] = `
<header
  className="root"
>
  <PageBlock>
    <div
      className="externalNav"
    >
      <CountryDropdown
        checked={0}
        linkRenderer={[Function]}
        links={
          Array [
            Object {
              "ItemIcon": [Function],
              "country": "hk",
              "language": "en",
              "title": "Hong Kong (English)",
              "url": "https://hk.jobsdb.com/hk",
            },
            Object {
              "ItemIcon": [Function],
              "country": "id",
              "language": "en",
              "title": "Indonesia (English)",
              "url": "https://id.jobsdb.com/id",
            },
          ]
        }
        messages={
          Object {
            "header.employerLinkPrefix": "Awesome employers should ",
            "header.employerSiteTitle": "click here",
            "header.employerSiteUrl": "http://brand.seekasia.com",
            "header.homeUrl": "http://seekasia.com/home",
            "header.homeUrlText": "awesome",
          }
        }
        onMenuToggle={[Function]}
      />
    </div>
    <div
      className="primaryNavNoLogin"
    >
      <h1
        className="logo"
      >
        <a
          className="logoLink"
          href="/"
        >
          <ScreenReaderOnly
            component="span"
          >
            awesome
          </ScreenReaderOnly>
        </a>
        <FakeComponent
          country="my"
        />
      </h1>
      <div
        className="primaryNavLinksWrapper"
      >
        <span
          className="primaryNavLinkWrapper"
          key="0"
        >
          <a
            className="primaryNavLink primaryNavLink"
            href="http://seekasia.com/home"
          >
            <Text
              baseline={true}
            >
               
              Home
            </Text>
          </a>
        </span>
        <span
          className="primaryNavLinkWrapper"
          key="1"
        >
          <a
            className="primaryNavLink primaryNavLink"
            href="http://seekasia.com/myPortal"
          >
            <Text
              baseline={true}
            >
               
              MyPortal
            </Text>
          </a>
        </span>
        <span
          className="primaryNavLinkWrapper"
          key="2"
        >
          <a
            className="primaryNavLink primaryNavLink"
            href="http://seekasia.com/resources"
          >
            <Text
              baseline={true}
            >
               
              Resources
            </Text>
          </a>
        </span>
        <span
          className="primaryNavLinkWrapper"
          key="3"
        >
          <a
            className="primaryNavLink primaryNavLink"
            href="http://seekasia.com/career-insights"
          >
            <Text
              baseline={true}
            >
               
              CareerInsights
            </Text>
          </a>
        </span>
      </div>
      <UserAccount
        authenticationStatus="unauthenticated"
        brandStyles={
          Object {
            "employerLink": "employerLink",
            "primaryNavLink": "primaryNavLink",
          }
        }
        linkRenderer={[Function]}
        loginAvailable={false}
        messages={
          Object {
            "header.employerLinkPrefix": "Awesome employers should ",
            "header.employerSiteTitle": "click here",
            "header.employerSiteUrl": "http://brand.seekasia.com",
            "header.homeUrl": "http://seekasia.com/home",
            "header.homeUrlText": "awesome",
          }
        }
        onMenuToggle={[Function]}
      />
    </div>
    <ActionTray
      activeTab="ACTIVE_TAB_HOME"
      brandStyles={
        Object {
          "employerLink": "employerLink",
          "primaryNavLink": "primaryNavLink",
        }
      }
      country="hk"
      handleToggleMenu={[Function]}
      linkRenderer={[Function]}
      menuOpen={false}
      messages={
        Object {
          "header.employerLinkPrefix": "Awesome employers should ",
          "header.employerSiteTitle": "click here",
          "header.employerSiteUrl": "http://brand.seekasia.com",
          "header.homeUrl": "http://seekasia.com/home",
          "header.homeUrlText": "awesome",
        }
      }
      showMenu={false}
    />
    <Menu
      authenticationStatus="unauthenticated"
      brandStyles={
        Object {
          "employerLink": "employerLink",
          "primaryNavLink": "primaryNavLink",
        }
      }
      linkRenderer={[Function]}
      links={
        Array [
          Object {
            "ItemIcon": [Function],
            "title": "Home",
            "url": "http://seekasia.com/home",
          },
          Object {
            "ItemIcon": [Function],
            "title": "MyPortal",
            "url": "http://seekasia.com/myPortal",
          },
          Object {
            "ItemIcon": [Function],
            "title": "Resources",
            "url": "http://seekasia.com/resources",
          },
          Object {
            "ItemIcon": [Function],
            "title": "CareerInsights",
            "url": "http://seekasia.com/career-insights",
          },
        ]
      }
      locales={
        Array [
          Object {
            "ItemIcon": [Function],
            "country": "hk",
            "language": "en",
            "title": "Hong Kong (English)",
            "url": "https://hk.jobsdb.com/hk",
          },
          Object {
            "ItemIcon": [Function],
            "country": "id",
            "language": "en",
            "title": "Indonesia (English)",
            "url": "https://id.jobsdb.com/id",
          },
        ]
      }
      loginAvailable={false}
      messages={
        Object {
          "header.employerLinkPrefix": "Awesome employers should ",
          "header.employerSiteTitle": "click here",
          "header.employerSiteUrl": "http://brand.seekasia.com",
          "header.homeUrl": "http://seekasia.com/home",
          "header.homeUrlText": "awesome",
        }
      }
      shouldShowMenu={false}
    />
  </PageBlock>
</header>
`;

exports[`Header should render with default props 1`] = `
<header
  className="root"
>
  <PageBlock>
    <div
      className="externalNav"
    >
      <CountryDropdown
        checked={0}
        linkRenderer={[Function]}
        links={
          Array [
            Object {
              "ItemIcon": [Function],
              "country": "hk",
              "language": "en",
              "title": "Hong Kong (English)",
              "url": "https://hk.jobsdb.com/hk",
            },
            Object {
              "ItemIcon": [Function],
              "country": "id",
              "language": "en",
              "title": "Indonesia (English)",
              "url": "https://id.jobsdb.com/id",
            },
          ]
        }
        messages={
          Object {
            "header.employerLinkPrefix": "Awesome employers should ",
            "header.employerSiteTitle": "click here",
            "header.employerSiteUrl": "http://brand.seekasia.com",
            "header.homeUrl": "http://seekasia.com/home",
            "header.homeUrlText": "awesome",
          }
        }
        onMenuToggle={[Function]}
      />
    </div>
    <div
      className="primaryNavNoLogin"
    >
      <h1
        className="logo"
      >
        <a
          className="logoLink"
          href="/"
        >
          <ScreenReaderOnly
            component="span"
          >
            awesome
          </ScreenReaderOnly>
        </a>
        <FakeComponent />
      </h1>
      <div
        className="primaryNavLinksWrapper"
      />
      <UserAccount
        authenticationStatus="unauthenticated"
        brandStyles={
          Object {
            "employerLink": "employerLink",
            "primaryNavLink": "primaryNavLink",
          }
        }
        linkRenderer={[Function]}
        loginAvailable={false}
        messages={
          Object {
            "header.employerLinkPrefix": "Awesome employers should ",
            "header.employerSiteTitle": "click here",
            "header.employerSiteUrl": "http://brand.seekasia.com",
            "header.homeUrl": "http://seekasia.com/home",
            "header.homeUrlText": "awesome",
          }
        }
        onMenuToggle={[Function]}
      />
    </div>
    <ActionTray
      brandStyles={
        Object {
          "employerLink": "employerLink",
          "primaryNavLink": "primaryNavLink",
        }
      }
      country="hk"
      handleToggleMenu={[Function]}
      linkRenderer={[Function]}
      menuOpen={false}
      messages={
        Object {
          "header.employerLinkPrefix": "Awesome employers should ",
          "header.employerSiteTitle": "click here",
          "header.employerSiteUrl": "http://brand.seekasia.com",
          "header.homeUrl": "http://seekasia.com/home",
          "header.homeUrlText": "awesome",
        }
      }
    />
    <Menu
      authenticationStatus="unauthenticated"
      brandStyles={
        Object {
          "employerLink": "employerLink",
          "primaryNavLink": "primaryNavLink",
        }
      }
      linkRenderer={[Function]}
      locales={
        Array [
          Object {
            "ItemIcon": [Function],
            "country": "hk",
            "language": "en",
            "title": "Hong Kong (English)",
            "url": "https://hk.jobsdb.com/hk",
          },
          Object {
            "ItemIcon": [Function],
            "country": "id",
            "language": "en",
            "title": "Indonesia (English)",
            "url": "https://id.jobsdb.com/id",
          },
        ]
      }
      loginAvailable={false}
      messages={
        Object {
          "header.employerLinkPrefix": "Awesome employers should ",
          "header.employerSiteTitle": "click here",
          "header.employerSiteUrl": "http://brand.seekasia.com",
          "header.homeUrl": "http://seekasia.com/home",
          "header.homeUrlText": "awesome",
        }
      }
      shouldShowMenu={false}
    />
  </PageBlock>
</header>
`;

exports[`Header should render with links list 1`] = `
<header
  className="root"
>
  <PageBlock>
    <div
      className="externalNav"
    >
      <CountryDropdown
        checked={0}
        linkRenderer={[Function]}
        links={
          Array [
            Object {
              "ItemIcon": [Function],
              "country": "hk",
              "language": "en",
              "title": "Hong Kong (English)",
              "url": "https://hk.jobsdb.com/hk",
            },
            Object {
              "ItemIcon": [Function],
              "country": "id",
              "language": "en",
              "title": "Indonesia (English)",
              "url": "https://id.jobsdb.com/id",
            },
          ]
        }
        messages={
          Object {
            "header.employerLinkPrefix": "Awesome employers should ",
            "header.employerSiteTitle": "click here",
            "header.employerSiteUrl": "http://brand.seekasia.com",
            "header.homeUrl": "http://seekasia.com/home",
            "header.homeUrlText": "awesome",
          }
        }
        onMenuToggle={[Function]}
      />
    </div>
    <div
      className="primaryNavNoLogin"
    >
      <h1
        className="logo"
      >
        <a
          className="logoLink"
          href="/"
        >
          <ScreenReaderOnly
            component="span"
          >
            awesome
          </ScreenReaderOnly>
        </a>
        <FakeComponent />
      </h1>
      <div
        className="primaryNavLinksWrapper"
      >
        <span
          className="primaryNavLinkWrapper"
          key="0"
        >
          <a
            className="primaryNavLink primaryNavLink"
            href="http://seekasia.com/home"
          >
            <Text
              baseline={true}
            >
               
              Home
            </Text>
          </a>
        </span>
        <span
          className="primaryNavLinkWrapper"
          key="1"
        >
          <a
            className="primaryNavLink primaryNavLink"
            href="http://seekasia.com/myPortal"
          >
            <Text
              baseline={true}
            >
               
              MyPortal
            </Text>
          </a>
        </span>
        <span
          className="primaryNavLinkWrapper"
          key="2"
        >
          <a
            className="primaryNavLink primaryNavLink"
            href="http://seekasia.com/resources"
          >
            <Text
              baseline={true}
            >
               
              Resources
            </Text>
          </a>
        </span>
        <span
          className="primaryNavLinkWrapper"
          key="3"
        >
          <a
            className="primaryNavLink primaryNavLink"
            href="http://seekasia.com/career-insights"
          >
            <Text
              baseline={true}
            >
               
              CareerInsights
            </Text>
          </a>
        </span>
      </div>
      <UserAccount
        authenticationStatus="unauthenticated"
        brandStyles={
          Object {
            "employerLink": "employerLink",
            "primaryNavLink": "primaryNavLink",
          }
        }
        linkRenderer={[Function]}
        loginAvailable={false}
        messages={
          Object {
            "header.employerLinkPrefix": "Awesome employers should ",
            "header.employerSiteTitle": "click here",
            "header.employerSiteUrl": "http://brand.seekasia.com",
            "header.homeUrl": "http://seekasia.com/home",
            "header.homeUrlText": "awesome",
          }
        }
        onMenuToggle={[Function]}
      />
    </div>
    <ActionTray
      brandStyles={
        Object {
          "employerLink": "employerLink",
          "primaryNavLink": "primaryNavLink",
        }
      }
      country="hk"
      handleToggleMenu={[Function]}
      linkRenderer={[Function]}
      menuOpen={false}
      messages={
        Object {
          "header.employerLinkPrefix": "Awesome employers should ",
          "header.employerSiteTitle": "click here",
          "header.employerSiteUrl": "http://brand.seekasia.com",
          "header.homeUrl": "http://seekasia.com/home",
          "header.homeUrlText": "awesome",
        }
      }
    />
    <Menu
      authenticationStatus="unauthenticated"
      brandStyles={
        Object {
          "employerLink": "employerLink",
          "primaryNavLink": "primaryNavLink",
        }
      }
      linkRenderer={[Function]}
      links={
        Array [
          Object {
            "ItemIcon": [Function],
            "title": "Home",
            "url": "http://seekasia.com/home",
          },
          Object {
            "ItemIcon": [Function],
            "title": "MyPortal",
            "url": "http://seekasia.com/myPortal",
          },
          Object {
            "ItemIcon": [Function],
            "title": "Resources",
            "url": "http://seekasia.com/resources",
          },
          Object {
            "ItemIcon": [Function],
            "title": "CareerInsights",
            "url": "http://seekasia.com/career-insights",
          },
        ]
      }
      locales={
        Array [
          Object {
            "ItemIcon": [Function],
            "country": "hk",
            "language": "en",
            "title": "Hong Kong (English)",
            "url": "https://hk.jobsdb.com/hk",
          },
          Object {
            "ItemIcon": [Function],
            "country": "id",
            "language": "en",
            "title": "Indonesia (English)",
            "url": "https://id.jobsdb.com/id",
          },
        ]
      }
      loginAvailable={false}
      messages={
        Object {
          "header.employerLinkPrefix": "Awesome employers should ",
          "header.employerSiteTitle": "click here",
          "header.employerSiteUrl": "http://brand.seekasia.com",
          "header.homeUrl": "http://seekasia.com/home",
          "header.homeUrlText": "awesome",
        }
      }
      shouldShowMenu={false}
    />
  </PageBlock>
</header>
`;

exports[`Header should render with login available 1`] = `
<header
  className="root"
>
  <PageBlock>
    <div
      className="externalNav"
    >
      <CountryDropdown
        checked={0}
        linkRenderer={[Function]}
        links={
          Array [
            Object {
              "ItemIcon": [Function],
              "country": "hk",
              "language": "en",
              "title": "Hong Kong (English)",
              "url": "https://hk.jobsdb.com/hk",
            },
            Object {
              "ItemIcon": [Function],
              "country": "id",
              "language": "en",
              "title": "Indonesia (English)",
              "url": "https://id.jobsdb.com/id",
            },
          ]
        }
        messages={
          Object {
            "header.employerLinkPrefix": "Awesome employers should ",
            "header.employerSiteTitle": "click here",
            "header.employerSiteUrl": "http://brand.seekasia.com",
            "header.homeUrl": "http://seekasia.com/home",
            "header.homeUrlText": "awesome",
          }
        }
        onMenuToggle={[Function]}
      />
    </div>
    <div
      className="primaryNav"
    >
      <h1
        className="logo"
      >
        <a
          className="logoLink"
          href="/"
        >
          <ScreenReaderOnly
            component="span"
          >
            awesome
          </ScreenReaderOnly>
        </a>
        <FakeComponent />
      </h1>
      <div
        className="primaryNavLinksWrapper"
      />
      <div
        className="secondaryNav"
      />
      <UserAccount
        authenticationStatus="unauthenticated"
        brandStyles={
          Object {
            "employerLink": "employerLink",
            "primaryNavLink": "primaryNavLink",
          }
        }
        linkRenderer={[Function]}
        loginAvailable={true}
        messages={
          Object {
            "header.employerLinkPrefix": "Awesome employers should ",
            "header.employerSiteTitle": "click here",
            "header.employerSiteUrl": "http://brand.seekasia.com",
            "header.homeUrl": "http://seekasia.com/home",
            "header.homeUrlText": "awesome",
          }
        }
        onMenuToggle={[Function]}
      />
    </div>
    <ActionTray
      brandStyles={
        Object {
          "employerLink": "employerLink",
          "primaryNavLink": "primaryNavLink",
        }
      }
      country="hk"
      handleToggleMenu={[Function]}
      linkRenderer={[Function]}
      menuOpen={false}
      messages={
        Object {
          "header.employerLinkPrefix": "Awesome employers should ",
          "header.employerSiteTitle": "click here",
          "header.employerSiteUrl": "http://brand.seekasia.com",
          "header.homeUrl": "http://seekasia.com/home",
          "header.homeUrlText": "awesome",
        }
      }
    />
    <Menu
      authenticationStatus="unauthenticated"
      brandStyles={
        Object {
          "employerLink": "employerLink",
          "primaryNavLink": "primaryNavLink",
        }
      }
      linkRenderer={[Function]}
      locales={
        Array [
          Object {
            "ItemIcon": [Function],
            "country": "hk",
            "language": "en",
            "title": "Hong Kong (English)",
            "url": "https://hk.jobsdb.com/hk",
          },
          Object {
            "ItemIcon": [Function],
            "country": "id",
            "language": "en",
            "title": "Indonesia (English)",
            "url": "https://id.jobsdb.com/id",
          },
        ]
      }
      loginAvailable={true}
      messages={
        Object {
          "header.employerLinkPrefix": "Awesome employers should ",
          "header.employerSiteTitle": "click here",
          "header.employerSiteUrl": "http://brand.seekasia.com",
          "header.homeUrl": "http://seekasia.com/home",
          "header.homeUrlText": "awesome",
        }
      }
      shouldShowMenu={false}
    />
  </PageBlock>
</header>
`;
